Website/src/rendering/src/components/Products/ProductDetailPage.tsx

```diff
@@ -0,0 +1,340 @@
+      ['Content-Type', 'application/json'],
+    ];
+
+    const url = 'https://api.rfksrv.com/search-rec/11269-125757327/3';
+
+    fetch(url, {
+      method: 'POST',
+      headers,
+      body: JSON.stringify({ data }),
+    })
+      .then((res) => res.json())
+      .then((res) => {
+        const similarProducts = getWidgetById(res, 'hs_sim');
+        const moreProducts = getWidgetById(res, 'hs_more');
+
+        setSimilarProducts(similarProducts?.content.product.value);
+        setMoreProducts(moreProducts?.content.product.value);
+      });
+  }, [sku]);
+
+  useEffect(() => {
+    if (sku) {
+      logViewEvent();
+      fetchData();
+    }
+  }, [sku, fetchData]);
+
+  const getWidgetById = (res: rfkResponse, id: string) => {
+    return res.batch.find((widget: Widget) => widget.widget.rfkid === id);
+  };
+
+  const productImage =
+    variant?.xp?.Images.length > 0 && variant?.xp?.Images[0].Url ? (
+      <img className="product-image-main" src={variant.xp.Images[0].Url} alt="variant" />
+    ) : product?.xp?.Images.length > 0 && product?.xp?.Images[0].Url ? (
+      <img className="product-image-main" src={product.xp.Images[0].Url} alt="product" />
+    ) : null;
+
+  // Force route to reload when navigating to a specific variant
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const goToVariant = (link: string) => {
+    window.location.href = link;
+  };
+  return product ? (
+    <section className="section">
+      <div className="section__content container">
+        <div className="product-detail">
+          <div className="product-detail-hero">
+            <div className="product-image">{productImage}</div>
+            <div className="product-description">
+              <h2>{product.Name}</h2>
+              <p>SKU: {sku}</p>
+              <div>{product.Description}</div>
+              <div>
+                Price:{' '}
+                <span className="product-price">{product.PriceSchedule.PriceBreaks[0].Price}</span>
+              </div>
+              <form onSubmit={handleAddToCart}>
+                {specs ? (
+                  <div>
+                    Specs:
+                    <div className="product-specs">
+                      {specs?.map((spec) => {
+                        const specValue = specValues.find(
+                          (specValue) => specValue.SpecID === spec.ID
+                        );
+                        return (
+                          <ProductSpecField
+                            key={spec.ID}
+                            spec={spec}
+                            onChange={handleSpecFieldChange}
+                            optionId={specValue && specValue.OptionID}
+                            value={specValue && specValue.Value}
+                          />
+                        );
+                      })}
+                    </div>
+                  </div>
+                ) : null}
+                {variants && !variantID ? (
```

Comment:
This line is another example of a condition inside the main JSX of the component.